apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs

```diff
@@ -276,5 +278,190 @@ private static string ValidateTableName(string table, string parameterName)
             }
             return table;
         }
+
+        private static readonly char s_singleQuote = '\'';
+        private static readonly char s_doubleQuote = '"';
+        private static readonly char s_backtickQuote = '`';
+        private static readonly char s_hyphen = '-';
+        private static readonly char s_dash = '#';
+        private static readonly char s_slash = '/';
+        private static readonly char s_asterisk = '*';
+
+        /// <summary>
+        /// Removes any comments from the given sql string.
+        /// </summary>
+        /// <param name="sql">The sql string to strip for comments</param>
+        /// <returns>The sql string without any comments</returns>
+        /// <exception cref="SpannerException">
+        /// Throws SpannerException if the sql string contains an unclosed
+        /// literal or unclosed quoted identifier
+        /// </exception>
+        internal static string RemoveComments(string sql)
+        {
+            GaxPreconditions.CheckNotNull(sql, nameof(sql));
+            bool isInQuoted = false;
+            bool isInSingleLineComment = false;
+            bool isInMultiLineComment = false;
+            char startQuote = '\0';
+            bool lastCharWasEscapeChar = false;
+            bool isTripleQuoted = false;
+            StringBuilder res = new StringBuilder(sql.Length);
+            int index = 0;
+            while (index < sql.Length)
+            {
+                char c = sql[index];
+                if (isInQuoted)
+                {
+                    if ((c == '\n' || c == '\r') && !isTripleQuoted)
+                    {
+                        throw new SpannerException(
```

Comment:
I think this being a parsing issue, it should throw the same exception as the `FromCommandText` method, which is `InvalidOperationException`. And applies to the rest of the exceptions here.

Reply:
I assume you mean `ArgumentException` (which is thrown by `FromCommandText`), so I've changed it to that.